sdk/keyvault/keyvault-keys/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -151,14 +151,11 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
         const { authorization, resource } = this.parseWWWAuthenticate(www_authenticate);
         const challenge = new AuthenticationChallenge(authorization, resource + "/.default")
 
-        if (!challenge.equalTo(this.challengeCache.challenge)) {
-          this.challengeCache.setCachedChallenge(challenge);
-          this.tokenCache.setCachedToken(undefined);
+        this.challengeCache.setCachedChallenge(challenge);
+        this.tokenCache.setCachedToken(undefined);
 
-          await this.authenticateRequest(webResource);
-          return this._nextPolicy.sendRequest(webResource);
-        }
-        return response;
+        await this.authenticateRequest(webResource);
+        return this._nextPolicy.sendRequest(webResource);
```

Comment:
What this fix does is that it assumes that if we reached the point where the server answered with 401, and with a valid Authenticate header, then we should always re-authenticate.